src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,36 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
+    private final int MINIMAL_AGE = 18;
+    private final int MINIMAL_LENGTH = 6;
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws ImpossibleRegisterUserException {
+        if (user == null) {
+            throw new ImpossibleRegisterUserException("The user can`t be empty");
+        } else if (user.getLogin() == null || user.getLogin().length() < MINIMAL_LENGTH) {
+            throw new ImpossibleRegisterUserException("This login is incorrect");
+        } else if (user.getPassword() == null || user.getPassword().length() < MINIMAL_LENGTH) {
+            throw new ImpossibleRegisterUserException("This password is incorrect");
```

Comment:
The exception message 'This password is incorrect' could be more informative. Specify whether it's due to being null or too short.